src/expression/index.ts

```diff
@@ -169,6 +169,7 @@ export class ZoomConstantExpression<Kind extends EvaluationKind> {
     isStateDependent: boolean;
     globalStateRefs: Set<string>;
     _styleExpression: StyleExpression;
+    globalState: Record<string, any> | null;
```

Comment:
Is this set only from the outside? I don't see how this is being used...
Should it be set in the constructor?

Reply:
No, there will only be one way of passing global state: by setting it on expression. It won't be passed as parameter to `evaluate` method.
However removing it from `evaluate` method now means a breaking change. My plan is to add it as an option here, change `maplibre-gl-js` to using set `globalState` on expression over a series of PR separately for layout, paint and filter properties and then remove the option of passing it as a parameter of `evaluate` here.

Reply:
In this case you could consider merging the PR [#6321](https://github.com/maplibre/maplibre-gl-js/pull/6321) as it fixes a known bug. I'll continue working on this more comprehensive fix.